### PR TITLE
✨ Modules & Commands

### PR DESCRIPTION
**Pull request by itsfolf**

### CODE REVIEW

File: src/main/kotlin/io/github/techbox/core/modules/commands/CommandRegistry.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import io.github.techbox.core.modules.ModuleRegistry.ModuleProxy
+import io.github.techbox.utils.logger
+import org.slf4j.Logger
+import java.util.*
+
+
+class CommandRegistry {
+    val log: Logger = logger<CommandRegistry>()
+    private val commands: HashMap<String, ICommand> = HashMap()
+    private val aliases: HashMap<String, String> = HashMap()
```
Comment (miguelmoraesp) [inline src/main/kotlin/io/github/techbox/core/modules/commands/CommandRegistry.kt:12]: Can you remove this and store the aliases in the command

**Review by miguelmoraesp — CHANGES_REQUESTED** [review]

File: src/main/kotlin/io/github/techbox/core/modules/commands/CommandRegistry.kt (hunk 1 above)
Comment (itsfolf) [reply src/main/kotlin/io/github/techbox/core/modules/commands/CommandRegistry.kt:12]: This map is used as a runtime cache to speed up aliases look up on command execution, the aliases are still stored and accessible inside the command.

File: src/main/kotlin/io/github/techbox/core/Techbox.kt [hunk 2]
```diff
@@ -111,7 +127,7 @@ class Techbox(private val config: Config) {
             shardStartListener.latch = CountDownLatch(latchCount)
             this.shardManager = shardManager.build()
 
-            threadPool.submit {
+            launch {
```
Comment (devnatan) [inline src/main/kotlin/io/github/techbox/core/Techbox.kt]: is important to know where the errors within the code come from during debugging.
Would be ideal to put a name for this coroutine, you can use it like this:
```kotlin
launch(CoroutineName("some name here")) {
   // ..
}
```

**Review by miguelmoraesp — APPROVED** [review]
LGTM!

**Pull request merged**